src/features/amUI/users/UsersList.tsx

```diff
@@ -24,12 +32,43 @@ export const UsersList = () => {
     if (!rightHolders) {
       return [[], 1];
     }
-    const numPages = getTotalNumOfPages(rightHolders, pageSize);
-    return [getArrayPage(rightHolders, currentPage, pageSize), numPages];
-  }, [rightHolders, currentPage]);
+
+    const filteredRightHolders: RightHolder[] = [];
+    rightHolders.forEach((rightHolder) => {
+      if (isSearchMatch(searchString, rightHolder)) {
+        filteredRightHolders.push(rightHolder);
+      } else if (rightHolder.inheritingRightHolders?.length > 0) {
+        // check for searchString matches in inheritingRightHolders
+        const matchingInheritingItems = rightHolder.inheritingRightHolders.filter(
+          (inheritRightHolder) => isSearchMatch(searchString, inheritRightHolder),
+        );
+        if (matchingInheritingItems.length > 0) {
+          filteredRightHolders.push({
+            ...rightHolder,
+            inheritingRightHolders: matchingInheritingItems,
+          });
+        }
+      }
+    });
+
+    const numPages = getTotalNumOfPages(filteredRightHolders, pageSize);
+    return [getArrayPage(filteredRightHolders, currentPage, pageSize), numPages];
+  }, [rightHolders, currentPage, searchString]);
+
+  const onSearch = (newSearchString: string) => {
+    setSearchString(newSearchString);
+    setCurrentPage(1); // reset current page when searching
+  };
 
   return (
     <div className={classes.usersList}>
+      <Search
+        className={classes.searchBar}
+        placeholder={t('users_page.user_search_placeholder')}
+        value={searchString}
+        onChange={(event) => onSearch(event.target.value)}
```

Comment:
Is it a good idea to debounce this? Continuous updates to the results list as the user types might be visually distracting and resource-intensive on the browser.

Reply:
I agree. We use debouncing on the other searches and even if this one doesn't call the BFF, there's no reason to use more resources than necessary